docs/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/update-folder-structure.md

```diff
@@ -0,0 +1,362 @@
+---
+title: v4 Plugin Migration - Update folder structure - Strapi Developer Docs
+description:
+canonicalUrl:
+sidebarDepth: 2
+---
+
+<!-- TODO: update SEO -->
+
+# v4 plugin migration: Update the folder structure
+
+!!!include(developer-docs/latest/update-migration-guides/migration-guides/v4/snippets/plugin-migration-intro.md)!!!
+
+::: callout
+Strapi v3 plugins required a specific folder structure.
+
+In Strapi v4, plugins are developed using a programmatic API, which gives flexibility in the folder structure.
+:::
+
+The folder structure of a Strapi v4 plugin should meet the following requirements:
+
+- The root of the plugin folder should include:
+  - a `strapi-server.js` entry file, if the plugin interacts with Strapi's back-end (see [Server API](/developer-docs/latest/setup-deployment-guides/configurations/optional/environment.md))
+  - a `strapi-admin.js` entry file, if the plugin interacts with Strapi's admin panel (see [Admin Panel API](/developer-docs/latest/setup-deployment-guides/configurations/optional/environment.md)).
+
+- `strapi-admin.js` and `strapi-server.js` should export the plugin's interface.
+
+As long as these requirements are met, the rest of the folder structure is up to you.
+
+::: details Example of a Strapi v4 plugin structure
+
+```jsx
+my-plugin
+├─ admin
+│  ├─ components
+│  ├─ pages
+│  ├─ // more folders and files
+│  └─ index.js
+├─ server
+│  ├─ config
+│  ├─ controllers
+│  ├─ middlewares
+│  ├─ policies
+│  ├─ routes
+│  ├─ bootstrap.js
+│  ├─ // more folders and files
+│  └─ index.js
+├─ strapi-admin.js // require('./admin')
+└─ strapi-server.js // require('./server')
+```
+
+:::
+
+::: note
+The whole back-end or front-end code of a plugin could be included in the `strapi-server` and `strapi-admin` entry files, but as your plugin code gets more complex, we recommend splitting files and using the folder structure given as an example in this guide.
```

Comment:
I feel as though this isn't really needed, yes it's possible but it's not a good practice for this exact reason. I suggest we change all instructions to follow how our CLI generates things. Those with higher level skills will understand what they can do differently if they want.

Reply:
I see what you are saying Derrick, I think I suggested adding something like this to drive home the idea that the folder structure we are recommending in this guide is a convention we follow but not a requirement. Just trying to emphasize the flexibility developers have. 

It is definitely not best practice so you are probably right we shouldn't mention something that might point someone in the wrong direction.

Reply:
@petersg83 suggested we add this :-)
However, indeed I agree this is not really something considered as best practice, and that's why I removed it after second thought.
Sorry Pierre. The goal of the documentation is not to try to cover every existing use case but to keep things as simple as possible while promoting best and/or recommended practices.

If people feel it is lacking, we'll re-add a note.